pkg/cmd/vendorapp.go

```diff
@@ -0,0 +1,136 @@
+package cmd
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/replicatedhq/embedded-cluster/pkg/metrics"
+	"github.com/replicatedhq/embedded-cluster/pkg/prompts"
+	"github.com/replicatedhq/embedded-cluster/pkg/release"
+	kotsv1beta1 "github.com/replicatedhq/kotskinds/apis/kots/v1beta1"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli/v2"
+)
+
+// maybePromptForAppUpdate warns the user if the embedded release is not the latest for the current
+// channel. If prompts are enabled, it will prompt the user to continue installing the out-of-date
+// release and return an error if the user chooses not to continue.
+func maybePromptForAppUpdate(c *cli.Context, prompt prompts.Prompt, license *kotsv1beta1.License) error {
+	// It is not possible to check for app updates in airgap mode.
+	if isAirgap := c.String("airgap-bundle") != ""; isAirgap {
+		return nil
+	}
+
+	channelRelease, err := release.GetChannelRelease()
+	if err != nil {
+		return fmt.Errorf("unable to get channel release: %w", err)
+	} else if channelRelease == nil {
+		// It is possible to install without embedding the release data. In this case, we cannot
+		// check for app updates.
+		return nil
+	}
+
+	if license == nil {
+		return errors.New("license required")
+	}
+
+	logrus.Debugf("Checking for pending app releases")
+
+	currentRelease, err := getCurrentAppChannelRelease(c.Context, license, channelRelease.ChannelID)
+	if err != nil {
+		return fmt.Errorf("get current app channel release: %w", err)
+	}
+
+	// In the dev and test environments, the channelSequence is set to 0 for all releases.
+	if channelRelease.VersionLabel == currentRelease.VersionLabel {
+		logrus.Debugf("Current app release is up-to-date")
+		return nil
+	}
+	logrus.Debugf("Current app release is out-of-date")
+
+	apiURL := metrics.BaseURL(license)
+	releaseURL := fmt.Sprintf("%s/embedded/%s/%s", apiURL, channelRelease.AppSlug, channelRelease.ChannelSlug)
+	logrus.Warnf("A newer version %s is available.", currentRelease.VersionLabel)
+	logrus.Infof(
+		"To download it, run:\n  curl -fL \"%s\" \\\n    -H \"Authorization: %s\" \\\n    -o %s-%s.tgz",
+		releaseURL,
+		license.Spec.LicenseID,
+		channelRelease.AppSlug,
+		channelRelease.ChannelSlug,
+	)
+
+	// if no-prompt is true, we don't prompt the user and continue by default.
+	// SKIP_APP_UPDATE_PROMPT is an escape hatch used by the CI to skip the prompt in case this
+	// release becomes out of date.
+	if !c.Bool("no-prompt") && os.Getenv("SKIP_APP_UPDATE_PROMPT") != "true" {
```

Comment:
doesn't the CI already use --no-prompt though? isn't `if !c.Bool("no-prompt") {` sufficient?